hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/StorageImpl.java

```diff
@@ -170,11 +170,14 @@ public MapKeysWithCursor fetchKeys(int tableIndex, int size) {
     public MapEntriesWithCursor fetchEntries(int tableIndex, int size, SerializationService serializationService) {
         List<Map.Entry<Data, R>> entries = new ArrayList<Map.Entry<Data, R>>(size);
         int newTableIndex = records.fetchEntries(tableIndex, size, entries);
-        List<Map.Entry<Data, Data>> entriesData = new ArrayList<Map.Entry<Data, Data>>(entries.size());
+        List<Map.Entry<Data, Object>> entriesData = new ArrayList<Map.Entry<Data, Object>>(entries.size());
         for (Map.Entry<Data, R> entry : entries) {
-            R record = entry.getValue();
-            Data dataValue = serializationService.toData(record.getValue());
-            entriesData.add(new AbstractMap.SimpleEntry<Data, Data>(entry.getKey(), dataValue));
+            R record = (R)(Record)entry.getValue();
+            Object value = record.getValue();
+            Object dataValue = (value instanceof com.hazelcast.map.Immutable) ?
```

Comment:
You can import this to avoid the fully-qualified class name. You can also change the variable name since this is no longer `dataValue`.